core/src/main/java/com/linecorp/armeria/common/multipart/MultipartDecoder.java

```diff
@@ -110,4 +121,92 @@ public void abort() {
     public void abort(Throwable cause) {
         decoded.abort(cause);
     }
+
+    @Override
+    public void onSubscribe(Subscription subscription) {
+        this.subscription = subscription;
+        assert subscriber != null;
+        subscriber.onSubscribe(new Subscription() {
+            @Override
+            public void request(long n) {
+                final long oldDemand = demand;
+                demand += n;
+                if (oldDemand == 0) {
+                    // We want first body publisher
+                    if (bodyPartPublisher == null) {
+                        //This will trigger DecodedHttpStreamMessage's upstream.
+                        subscription.request(1);
+                    } else {
+                        // Just wait bodyPart finish.
+                    }
+                }
+            }
+
+            @Override
+            public void cancel() {
+                subscription.cancel();
+            }
+        });
+    }
+
+    @Override
+    public void onNext(BodyPart bodyPart) {
+        assert subscriber != null;
+        demand--;
+        subscriber.onNext(bodyPart);
+    }
+
+    @Override
+    public void onError(Throwable t) {
+        assert subscriber != null;
+        subscriber.onError(t);
+    }
+
+    @Override
+    public void onComplete() {
+        assert subscriber != null;
+        subscriber.onComplete();
+    }
+
+    BodyPartPublisher onBodyPartBegin() {
```

Comment:
Any idea of this part. I want to make MultipartDecoder can know BodyPartPublisher and let BodyPartPublisher can call `decoded.askUpstreamForElement();`.